src/server/pps/server/api_server.go

```diff
@@ -2492,6 +2496,18 @@ func (a *apiServer) StartPipeline(ctx context.Context, request *pps.StartPipelin
 		return nil, err
 	}
 
+	// Remove 'Stopped' from the pipeline spec
+	// TODO(msteffen): can I get rid of this and have the PPS master make the
+	// change?
```

Comment:
It seems unlikely that you would be able to get rid of this, since there needs to be some change that triggers the master to start the pipeline and this is that change. Unless I'm misunderstanding something here.

Reply:
Removed this todo. My idea was that we'd be able to removed `Stopped` from the pipeline spec and rely on the pipeline state in etcd (which I actually still think might be possible—we'd need `CreatePipeline` to not change the pipeline state during updates, so that if a pipeline's in `STOPPED` and its spec commit changes, we update the RC without taking it out of `STOPPED`). But that's more than just changing StartPipeline